### PR TITLE
feat: websocket events test

### PR DESCRIPTION
**Pull request by lljxx1**

### CODE REVIEW

File: packages/indexer/package.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
     "debug": "yarn build && yarn migrate up; node --inspect -r module-alias/register ./dist/index.js",
     "format": "prettier --write \"src/**/*.+(js|ts|json)\"",
     "cli": "node -r module-alias/register ./dist/cli/index.js",
-    "test": "jest --no-watchman"
+    "test": "jest --no-watchman",
+    "ci-test": "jest --no-watchman src/tests/websocket/**.test.ts"
```
Comment (nofir) [inline packages/indexer/package.json:18]: is that just for local dev?

File: packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-job.ts [hunk 2]
```diff
@@ -33,162 +33,152 @@ export class SaleWebsocketEventsTriggerQueueJob extends AbstractRabbitMqJobHandl
     delay: 1000,
   } as BackoffStrategy;
 
-  protected async process(payload: SaleWebsocketEventsTriggerQueueJobPayload) {
+  static async format(payload: SaleWebsocketEventsTriggerQueueJobPayload) {
```
Comment (nofir) [inline packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-job.ts:36]: Just to i can understand, what is the reasoning for needing that?

can't you initiate a new instance of SaleWebsocketEventsTriggerQueueJob and call the process when you do the test?

File: packages/indexer/src/tests/websocket/sale.test.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    //     if (typeof value === "string" && value.includes("=")) {
+    //       (payload as any)[key] = fromBuffer(Buffer.from(value, "base64"));
+    //     } else if (value !== null && typeof value === "object") {
+    //       fixBase64(payload[key]);
+    //     }
+    //   }
+    // }
+    // fixBase64(payload);
+
+    const message = await SaleWebsocketEventsTriggerQueueJob.format({
```
Comment (nofir) [inline packages/indexer/src/tests/websocket/sale.test.ts:57]: following my question about the need for format, wouldn't it be possible to assert `toBeCalledWith` on publishWebsocketEvent?

I think it has the same outcome but this approach looks what goes in and out of the process.

File: packages/indexer/src/tests/websocket/sale.test.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      data: payload,
+    } as unknown as SaleWebsocketEventsTriggerQueueJobPayload);
+
+    const status = JoiSale.validate(message!.data);
+
+    expect(status.error).toBe(undefined);
+    expect(message!.event).toBe("sale.created");
+    expect(message!.tags.contract).toBe("0x4e76c23fe2a4e37b5e07b5625e17098baab86c18");
+    expect(message!.tags.maker).toBe("0xd805cec34482ee455144dd04aad09f2758cddfe8");
+    expect(message!.tags.taker).toBe("0x98415e65216c83910f82f663dc152a535c0428b5");
```
Comment (nofir) [inline packages/indexer/src/tests/websocket/sale.test.ts:67]: I assume this is incomplete

File: packages/indexer/package.json (hunk 1 above)
Comment (lljxx1) [reply packages/indexer/package.json:18]: we could use this to run the tests in CI

File: packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-job.ts (hunk 2 above)
Comment (lljxx1) [reply packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-job.ts]: addressed, but `process` is protected, it's can't be called outside. 
so I added `handle(payload: any)` to AbstractRabbitMqJob Handler to process the payload synchronized

**Review by nofir — APPROVED** [review]
looks good. left a comment with a question

Also, because of the change in code, lets just make sure we merge it to development and angel can confirm there is not issue with sale ws events

File: packages/indexer/src/jobs/abstract-rabbit-mq-job-handler.ts [hunk 5]
```diff
@@ -174,6 +174,10 @@ export abstract class AbstractRabbitMqJobHandler {
     }
   }
 
+  public async handle(payload: any) {
```
[the remark this answers is not in the document]
Comment (lljxx1) [reply packages/indexer/src/jobs/abstract-rabbit-mq-job-handler.ts]: @ipeleg OK!

**Pull request merged**